[16.0][ADD] product_packaging_level_sellable

QuocDuong1306 opened the pull request:
This module allows odoo users to set:

* Can be sold: On product packaging type model, this checkbox defines if product
  packagings from this particular type are available to be selected on sale
  order line.

damdam-s committed:
[ADD] module 'product_packaging_type' and 'product_packaging_type_required'

gurneyalex committed:
[FIX] provide a default value for the new required field

gurneyalex committed:
[FIX] sequence must be stored

grindtildeath committed:
Add qty per package type computed field

grindtildeath committed:
Improvements according to review

grindtildeath committed:
Fix division by zero error

Commit:
Migrate product_packaging_type to 13.0

Commit:
run pre-commit with new prettiers

oca-travis committed:
[UPD] Update product_packaging_type.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

Commit:
product_packaging_type: switch to LGPL

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
product_packaging_type 13.0.1.0.1

MiquelRForgeFlow committed:
[ADD] product_abc_classification

Tonow-c2c committed:
[IMP][product_packaging_type] Specific UNECE product packaging naming.

**Context**

* Manage UNECE codes on packaging
* Be able to express a packaging qty in smallest packaging qty
* Simplify packaging name management to ease translation and avoid people to write a name every time they add a packaging on a product

**Usage**

When I open a product form, in the inventory tab I want to see the product packaging and its related code I can quickly see the Qty per packaging to know how much of a given packaging is contain in another

**TODO**

* Update (Improve) the module "product_packaging_type" from product-attribute (I think ) to:
    * in the product "Inventory" tab form (Packaging list view)
        * Hide the name of the packaging
        * Move the "Packaging type" in the first position
    * When opening the packaging form view from  "Inventory" tab of the product
        * Replace the name by packaging type at the top
        * Hide name field, but populate it with on_change of packaging with the packaging type name
    * Name of the packaging type:
        * name_get should be: "NAME + ' ' + (CODE)" -> e.g. Retail box (HV)
        * name_get of the product packaging should be overridden:

```python
            if product_id is set:
                return packaging_type name (e.g. Transport box)
            else:
                return the recorded packaging name on the packaging (e.g. DHL box)
```

oca-travis committed:
[UPD] Update product_packaging_type.pot

OCA-git-bot committed:
product_packaging_type 13.0.1.1.0

Commit:
Add unique constraint on Packaging Type

The constraint was added by module
OCA/stock-logistics-warehouse::stock_cubiscan, which makes no sense, see:
https://github.com/OCA/stock-logistics-warehouse/blob/95cc85c470dcdf6f41cec828e39beac4a9786c80/stock_cubiscan/models/stock.py#L15-L26

We should have only one type of packaging per product, but there is a
twist for the default packaging.

As the packaging type is required on a package, to be a good player with
other modules and existing data, a default packaging type is created,
and is used as a default value.

An existing database with 4 packaging per product will have the default
on all the packagings. So the unique constraint ignores the default
type, allowing users to transition to the proper types.

Commit:
Set noupdate on default packaging type

Tonow-c2c committed:
[IMP] If Product packaging "Contained quantity" == 0.0 "Has GTIN" is False

In product packaging form view the "Has GTIN" is invisible

Tonow-c2c committed:
[IMP] rename the field `type_has_gtin` to `barcode_required_for_gtin`

oca-travis committed:
[UPD] Update product_packaging_type.pot

OCA-git-bot committed:
product_packaging_type 13.0.1.1.1

Commit:
Update product_packaging_type/models/product_packaging_type.py

Commit:
Replace SQL constraint for unique packaging type by Python constrains

The SQL constraint I added previously [0] prevent some demo data to
be installed (e.g. in module OCA/wms::stock_storage_type).

The issue is that the packaging type is required, is part of the unique
constraint, and has a default packaging type.
When the module is installed and we have existing data or demo data,
the default type will be set on all the packagings, meaning we have
to ignore the unique constraint on the default type.
This was already handled using an "EXCLUDE" constraint, but it does
not solve the issue with demo data, the related field
"packaging_type_is_default", used as exclusion for the unique
constraint, is first inserted with a NULL value before being set
to True.

[0] https://github.com/OCA/product-attribute/pull/654

oca-travis committed:
[UPD] Update product_packaging_type.pot

OCA-git-bot committed:
product_packaging_type 13.0.1.2.0

oca-travis committed:
[UPD] Update product_packaging_type.pot

OCA-git-bot committed:
product_packaging_type 13.0.1.2.1

claudiagn committed:
Added translation using Weblate (Spanish)

claudiagn committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (25 of 25 strings)

Translation: product-attribute-13.0/product-attribute-13.0-product_packaging_type
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_packaging_type/es/

phuctranfxvn committed:
[MIG] product_packaging_type: Migration to 14.0

lmarion-source committed:
[IMP] product_abc_classification: Makes the classification computation logic pluggable

lmarion-source committed:
[IMP] product_abc_classification: Makes the classification computation logic pluggable

lmignon committed:
[IMP] product_abc_classification: Add sale_stock classification profile
This new profile provides a computation method based on the number of sale order line delivered by products

lmignon committed:
[IMP] product_abc_classification: Add sale_stock classification profile
This new profile provides a computation method based on the number of sale order line delivered by products

lmignon committed:
[IMP] product_abc_classification_sale_stock: Store collected data used to compute the classification

lmignon committed:
[IMP] product_abc_classification_sale_stock: Store collected data used to compute the classification

lmignon committed:
[IMP] product_abc_classification: Take into a account the expected distribution of produts

lmignon committed:
[IMP] product_abc_classification: Take into a account the expected distribution of produts

lmignon committed:
[FIX] product_abc_classification_base: Fix bug in views

oca-travis committed:
[UPD] Update product_packaging_type.pot

OCA-git-bot committed:
[UPD] README.rst

bosd committed:
Added translation using Weblate (Dutch)

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 92.0% (23 of 25 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_packaging_type
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_packaging_type/nl/

simahawk committed:
p_packaging_type: imp qty per type

* make computing func reusable
* drop hardcode html formatting

The html formatting is not required and can be changed on demand now.

simahawk committed:
product_packaging_type: split qty per type data

Makes qty per type data reusable from outside.

OCA-git-bot committed:
product_packaging_type 14.0.0.2.0

ChrisOForgeFlow committed:
[15.0][MIG] product_packaging_type

ChrisOForgeFlow committed:
[FIX] pre-commit

lmarion-source committed:
[IMP] Add flag to automatically set manual value of level_id to computed one

lmarion-source committed:
[IMP] Add flag to automatically set manual value of level_id to computed one

lmarion-source committed:
[FIX] Trigger the auto compute if flag changes but not the computed_value

lmarion-source committed:
[FIX] Trigger the auto compute if flag changes but not the computed_value

lmarion-source committed:
[FIX] Set flag to false if manual = computed

MiquelRForgeFlow committed:
[IMP] product_abc_classification: add product smart button in profile

lmarion-source committed:
[FIX] Propagate recompute level info to all computed values (abc storage on product, level_id, flag, ...)

Commit:
[UPD] Update product_packaging_type.pot

OCA-git-bot committed:
[UPD] README.rst

pablontura committed:
Added translation using Weblate (Catalan)

pablontura committed:
Translated using Weblate (Catalan)

Currently translated at 96.0% (24 of 25 strings)

Translation: product-attribute-15.0/product-attribute-15.0-product_packaging_type
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-15-0/product-attribute-15-0-product_packaging_type/ca/

rousseldenis committed:
[IMP] product_abc_classification_base: pre-commit stuff

rousseldenis committed:
[MIG][16.0] product_abc_classification_base

rousseldenis committed:
[IMP] product_abc_classification_base: Remove web_m2x_options dependency

rousseldenis committed:
[IMP] product_abc_classification/base: Restore module name

rousseldenis committed:
[FIX] product_abc_classification: Remove not working context attributes

rousseldenis committed:
[IMP] product_abc_classification_sale_stock: pre-commit stuff

rousseldenis committed:
[MIG][16.0] product_abc_classification_sale_stock

rousseldenis committed:
[FIX] product_abc_classification: Use the good _for_xml_id() and add tests

rousseldenis committed:
[FIX] product_abc_classification: Adapt write() for a multi recordset

rousseldenis committed:
[IMP] product_packaging_type: pre-commit stuff

rousseldenis committed:
[MIG][16.0] product_packaging_type

rousseldenis committed:
[IMP] product_packaging_type: Rename to product_packaging_level

rousseldenis committed:
[IMP] product_packaging_level: Add migration scripts

rousseldenis committed:
[FIX] product_abc_classification_sale_stock: Don't take into account results if void

rousseldenis committed:
[IMP] product_abc_classification: Improve profile view + help

lmignon committed:
[FIX] product_abc_classification: Translated terms into views

lmignon committed:
[FIX] product_abc_classification_sale_stock: Translated terms into views

hparfr committed:
product_sale_description: add module

Commit:
[ADD] product_packaging_level_salable

nilshamerlinck commented on product_packaging_level_sellable/__manifest__.py:
```diff
@@ -0,0 +1,20 @@
+# Copyright 2020 Camptocamp SA
```
2023

nilshamerlinck commented on product_packaging_level_sellable/models/product_packaging.py:
```diff
@@ -0,0 +1,16 @@
+# Copyright 2020 Camptocamp SA
```
2023

nilshamerlinck commented on product_packaging_level_sellable/models/sale_order_line.py:
```diff
@@ -0,0 +1,51 @@
+
+
+class SaleOrderLine(models.Model):
+
```
```suggestion
```

nilshamerlinck commented on product_packaging_level_sellable/readme/CONFIGURE.rst:
```diff
@@ -0,0 +1,5 @@
+This module allows odoo users to set:
+
+* Can be sold: On product packaging type model, this checkbox defines if product
+  packagings from this particular type are available to be selected on sale
```
```suggestion
* Can be sold: On product packaging level model, this checkbox defines if product
  packagings from this particular level are available to be selected on sale
```

nilshamerlinck commented on product_packaging_level_sellable/tests/common.py:
```diff
@@ -0,0 +1,91 @@
+        cls.product = cls.env.ref("product.product_product_9")
+
+    @classmethod
+    def setUpClassPackagingType(cls):
```
```suggestion
    def setUpClassPackagingLevel(cls):
```

nilshamerlinck commented on product_packaging_level_sellable/tests/test_packaging_level_can_be_sold.py:
```diff
@@ -0,0 +1,49 @@
+            self.order_line.write(
+                {"product_packaging_id": self.packaging_cannot_be_sold.id}
+            )
+        # Packaging can be sold even if the packaging type does not allows it
```
```suggestion
        # Packaging can be sold even if the packaging level does not allows it
```

nilshamerlinck reviewed (changes requested):
I think you should also add a new column can be sold in the tree view

QuocDuong1306 commented:
Hi @nilshamerlinck , I updated

Commit:
[IMP] product_packaging_level_salable: Add migration scripts

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (27 of 27 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (2 of 2 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_multi_category
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_multi_category/es/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (34 of 34 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_expiry_configurable
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_expiry_configurable/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (7 of 7 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_cost_security
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_cost_security/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (12 of 12 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_dimension
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_dimension/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (28 of 28 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_logistics_uom
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_logistics_uom/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (6 of 6 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_manufacturer
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_manufacturer/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (7 of 7 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_net_weight
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_net_weight/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (65 of 65 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_direct_print
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_direct_print/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (22 of 22 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_simulation
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_simulation/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (7 of 7 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_simulation_margin
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_simulation_margin/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (14 of 14 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_supplierinfo
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_supplierinfo/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (24 of 24 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_secondary_unit/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (16 of 16 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_sequence
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_sequence/es/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (19 of 19 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_direct_print_xlsx
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_direct_print_xlsx/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (53 of 53 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_supplierinfo_for_customer
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_supplierinfo_for_customer/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (23 of 23 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_set
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_set/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (33 of 33 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_state
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_state/es/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (14 of 14 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_attribute_company_favorite
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_attribute_company_favorite/es/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (7 of 7 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_category_type
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_category_type/es/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (12 of 12 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_uom_measure_type
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_uom_measure_type/es/

dcorio committed:
Added translation using Weblate (Italian)

dcorio committed:
Translated using Weblate (Italian)

Currently translated at 84.6% (55 of 65 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_direct_print
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_direct_print/it/

dcorio committed:
Added translation using Weblate (Italian)

dcorio committed:
Translated using Weblate (Italian)

Currently translated at 94.7% (18 of 19 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_direct_print_xlsx
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_direct_print_xlsx/it/

OCA-git-bot committed:
Merge PR #1349 into 16.0

Signed-off-by sebastienbeau

Commit:
[UPD] Update product_sale_description.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
product_sale_description 16.0.1.0.1

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 84.6% (55 of 65 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_direct_print
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_direct_print/it/

sergio-teruel committed:
[FIX] product_secondary_unit: wrong factor computation for uom's equal between secondary uom and product uom

OCA-git-bot committed:
Merge PR #1415 into 16.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
product_secondary_unit 16.0.1.0.1

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

rousseldenis committed:
[FIX] product_abc_classification: Fix tests

OCA-git-bot committed:
Merge PR #1192 into 16.0

Signed-off-by rousseldenis

Commit:
[UPD] Update product_abc_classification.pot

OCA-git-bot committed:
oca-github-bot post-merge updates

rousseldenis committed:
[IMP] product_packaging_level: Improve README

OCA-git-bot committed:
Merge PR #1196 into 16.0

Signed-off-by rousseldenis

Commit:
[UPD] Update product_abc_classification_sale_stock.pot

OCA-git-bot committed:
oca-github-bot post-merge updates

OCA-git-bot committed:
Merge PR #1215 into 16.0

Signed-off-by rousseldenis

Commit:
[UPD] Update product_packaging_level.pot

OCA-git-bot committed:
oca-github-bot post-merge updates
